trappy/plot_utils.py

```diff
@@ -140,6 +140,114 @@ def number_freq_plots(runs, map_label):
 
     return num_freq_plots
 
+def plot_generic(dfr, name_field, names=None, fields=None,
+                 prettify_name=None, width=None, height=None, xlim="default",
+                 ylim="range", drawstyle="default", ax=None, title=""):
```

Comment:
So, on the `plot_generic` thing, it's a good idea but it duplicates some logic in trappy, namely the grammar use by `trappy.stats.grammar.Parser` and `trappy.ILinePlot`. 

```
plot_generic(thermal_dfr, "cpus", fields=["cdev_state"])
```

would roughly (I haven't actually tested this) translate to

```
ILinePlot(thermal_dfr, column="cdev_state", pivot="cpus")
```

This actually does something totally different because it only works for the interactive plots that can be used in IPython, not matplotlib. So what we ideally want is probably to extend that existing code to work with matplotlib instead. That might be a bit of a lofty goal (not sure), but we should definitely make the two interfaces use the same terminology and if possible, design the interface so that it _looks like_ the code is shared :P 

What you've called a "determinant" is called a "pivot" in the existing code, and what you've called "names" is implemented with "filters". So if you just want to plot CPU 0's load you do `ILinePlot(sched_load_avg_cpu_df, column='util_avg', pivot='cpu', filters={'cpu': [0]})`.

But this is probably too much detail, we should first finish the discussion started by @derkling about where plotting code belongs, I think.